pkg/aws/g8scontrolplane/admit_g8scontrolplane_replicas_test.go

```diff
@@ -0,0 +1,125 @@
+package g8scontrolplane
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/giantswarm/admission-controller/pkg/admission"
+	"github.com/giantswarm/admission-controller/pkg/unittest"
+)
+
+func TestReplicasG8sControlPlaneAdmit(t *testing.T) {
+	testCases := []struct {
+		name                    string
+		ctx                     context.Context
+		currentAvailabilityZone []string
+		currentReplicas         int
+		expectReplicas          int
+	}{
+		{
+			name: "case 0",
+			ctx:  context.Background(),
+
+			currentAvailabilityZone: []string{"eu-central-1a"},
+			currentReplicas:         0,
+			expectReplicas:          1,
+		},
+		{
+			name: "case 1",
+			ctx:  context.Background(),
+
+			currentAvailabilityZone: []string{"eu-central-1a", "eu-central-1b", "eu-central-1c"},
+			currentReplicas:         0,
```

Comment:
What happens if we have 1 or 3 current replicas though? These tests are missing IMO.

Reply:
case 3 covers the case where we have 1 current replica. But I can add the one with 3 current replicas as well.